utils/gear-replay-cli/src/cmd/replay_block.rs

```diff
@@ -0,0 +1,179 @@
+    #[arg(short, long, num_args = 1..)]
+    pallet: Vec<String>,
+
+    /// Fetch the child-keys as well.
+    ///
+    /// Default is `false`, if specific `--pallets` are specified, `true` otherwise. In other
+    /// words, if you scrape the whole state the child tree data is included out of the box.
+    /// Otherwise, it must be enabled explicitly using this flag.
+    #[arg(long)]
+    child_tree: bool,
+}
+
+pub(crate) async fn replay_block<Block>(
+    shared: SharedParams,
+    command: ReplayBlockCmd<Block>,
+) -> sc_cli::Result<()>
+where
+    Block: BlockT + DeserializeOwned,
+    Block::Header: DeserializeOwned,
+    Block::Hash: FromStr,
+    <Block::Hash as FromStr>::Err: Debug,
+{
+    // Initialize the RPC client.
+    // get the block number associated with this block.
+    let block_ws_uri = shared.uri.clone();
+    let rpc = ws_client(&block_ws_uri).await?;
+
+    let block_hash_or_num = match command.block.clone() {
+        Some(b) => b,
+        None => {
+            let height = ChainApi::<
+                (),
+                <Block as BlockT>::Hash,
+                <Block as BlockT>::Header,
+                SignedBlock<Block>,
+            >::finalized_head(&rpc)
+            .await
+            .map_err(rpc_err_handler)?;
+
+            log::info!(
+                target: LOG_TARGET,
+                "Block is not provided, setting it to the latest finalized head: {:?}",
+                height
+            );
+
+            BlockHashOrNumber::Hash(height)
+        }
+    };
+
+    let (current_number, current_hash) = match block_hash_or_num {
+        BlockHashOrNumber::Number(n) => (n, block_number_to_hash::<Block>(&rpc, n).await?),
+        BlockHashOrNumber::Hash(hash) => (block_hash_to_number::<Block>(&rpc, hash).await?, hash),
+    };
+
+    // Get the state at the height corresponging to previous block.
+    let previous_hash =
+        block_number_to_hash::<Block>(&rpc, current_number.saturating_sub(One::one())).await?;
+    log::info!(
+        target: LOG_TARGET,
+        "Fetching state from {:?} at {:?}",
+        shared.uri,
+        previous_hash,
+    );
+    let ext = build_externalities::<Block>(
+        shared.uri.clone(),
+        Some(previous_hash),
+        command.pallet.clone(),
+        command.child_tree,
+    )
+    .await?;
+
+    log::info!(target: LOG_TARGET, "Fetching block {:?} ", current_hash);
+    let block = fetch_block::<Block>(&rpc, current_hash).await?;
+
+    // A digest item gets added when the runtime is processing the block, so we need to pop
+    // the last one to be consistent with what a gossiped block would contain.
+    let (mut header, extrinsics) = block.deconstruct();
+    header.digest_mut().pop();
+    let block = Block::new(header, extrinsics);
+
+    // Create executor, suitable for usage in conjunction with the preferred execution strategy.
+    #[cfg(all(not(feature = "always-wasm"), feature = "vara-native"))]
```

Comment:
do we panic in cases of vara and gear native at the same time

Reply:
Well, eventually it will panic if the executor doesn't match the actual runtime being used.
I don't think we should be over-protective here as this is not a production code and we rely on the caller to know what they are doing.

Reply:
I'd add comment about this for such places only then

Reply:
On second thoughts, we shouldn't even have a panic in such case: if the native runtime version (expected by the actual native executor) doesn't match the onchain version from the downloaded runtime, the "fallback" WASM executor will be used.